src/Kernel/Magic/QirMagic.cs

```diff
@@ -56,21 +98,51 @@ public class QirMagic : AbstractMagic
         ISymbolResolver symbolResolver,
         IConfigurationSource configurationSource,
         IMetadataController metadataController,
+        IEventService eventService,
         ISnippets snippets
     ) : base(
         "qir",
         new Microsoft.Jupyter.Core.Documentation
         {
             Summary = "Compiles a given Q# entry point to QIR, saving the resulting QIR to a given file.",
-            Description = @"
+            Description = $@"
                 This command takes the full name of a Q# entry point, and compiles the Q# from that entry point
                 into QIR. The resulting program is then executed, and the output of the program is displayed.
 
                 #### Required parameters
 
-                - Q# operation or function name. This must be the first parameter, and must be a valid Q# operation
+                - `{ParameterNameOperationName}=<string>`: Q# operation or function name.
+                This must be the first parameter, and must be a valid Q# operation
                 or function name that has been defined either in the notebook or in a Q# file in the same folder.
-                - The file path for where to save the output QIR to, specified as `output=<file path>`.
+
+                #### Optional parameters
+
+                - `{ParameterNameTarget}=<string>`: The intended execution target for the compiled entrypoint.
+                Defaults to the active Azure Quantum target (which can be set with `%azure.target`).                
+                Otherwise, defaults to a generic target, which may not work when running on a specific target.
+
+                - `{ParameterNameTargetCapability}=<string>`: The capability of the intended execution target.
+                If `{ParameterNameTarget}` is specified or there is an active Azure Quantum target,
+                defaults to the target's maximum capability.
+                Otherwise, defaults to `FullComputation`, which may not be supported when running on a specific target.
+                Possible options are:
+                    * `{TargetCapabilityModule.Top}`
+                    * `{TargetCapabilityModule.Bottom}`
```

Comment:
Is this surfaced to the user? What is the intent of `TargetCapabilityModule.Top` and `TargetCapabilityModule.Bottom`.

Reply:
As discussed, I removed the list of possible `TargetCapability` from the comments, and moved the comments about the new non-user facing parameters to the developer documentation instead of the `%qir` magic command documentation.